### ISSUE TITLE
Facing issue in the new version 3.22.2 while publishing to web

### ISSUE THREAD

**Issue by mriduljainmj** [FlutterFlow/flutterflow-issues#3229]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Before turning off CanvasKit.
Error: Command failed: flutter build web --web-renderer canvaskit --no-pub --no-version-check 2>&1

Before turning off CanvasKit.
Error: Command failed: flutter build web --web-renderer html --no-pub --no-version-check --no-tree-shake-icons 2>&1

### Expected Behavior

It should get publish on web

### Steps to Reproduce

1). Create a simple project
2). Try to publish it on the web

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITFThcn1z5d1obhe+KrucMJV+WYvGUMfWOdEket+EA0hfbmrELMQdcnOR1tLYteod1VIfFGWmTgG0d7QkYXlF/UEBwutRbp/zq5XQQLMJU2hSqayDKieWnNSQMJmGkOBy6a4kQhCItdYLWQFwV2IHPCdG3qCf9qOYwx5e6fDbOY=

### Visual documentation

When I try to publish the project on web it is throwing an error  "Error: Command failed: flutter build web --web-renderer canvaskit --no-pub --no-version-check 2>&1"
The version is latest 3.22.2

### Environment

```markdown
- FlutterFlow version: 3.22.2
- Platform: Web
- Browser name and version: Chrome 124.0.6367.158
- Operating system and version affected: MacOs 14.5 (23F79)
```

**Comment by rzambroni** [comment]
Hey @mriduljainmj, sorry about that issue, it looks like it was caused by the macOS app being a bit behind. I'm marking this as a duplicate in favor of a previous ticket: [https://github.com/FlutterFlow/flutterflow-issues/issues/3225](https://github.com/FlutterFlow/flutterflow-issues/issues/3225)

Please refer to that ticket for updates.

Thanks.

**Comment by ValetCEO** [comment]
Same issuse here - Still can't deploy to web

**Comment by rsstevearend** [comment]
Same issue can't publish to the web --- I'm on mac.  Here is the error 
Warning: In index.html:67: Local variable for "serviceWorkerVersion" is deprecated. Use "{{flutter_service_worker_version}}" template token instead.
Warning: In index.html:82: "FlutterLoader.loadEntrypoint" is deprecated. Use "FlutterLoader.load" instead.
Waiting for connection from debug service on Chrome...             28.7s
This app is linked to the debug service: ws://127.0.0.1:53655/WA_dlSCeZmo=/ws
Debug service listening on ws://127.0.0.1:53655/WA_dlSCeZmo=/ws

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".

A Dart VM Service on Chrome is available at: http://127.0.0.1:53655/WA_dlSCeZmo=
Rejecting promise with error: TypeError: Cannot set properties of undefined (setting 'nativeCommunication')


Performing hot restart...                                       
Warning: In index.html:67: Local variable for "serviceWorkerVersion" is deprecated. Use "{{flutter_service_worker_version}}" template token instead.
Warning: In index.html:82: "FlutterLoader.loadEntrypoint" is deprecated. Use "FlutterLoader.load" instead.
Performing hot restart...                                          22.5s
Restarted application in 22,459ms.
Rejecting promise with error: TypeError: Cannot set properties of undefined (setting 'nativeCommunication')

**Comment by ValetCEO** [comment]
<img width="1680" alt="Screenshot 2024-06-26 at 09 53 54" src="https://github.com/FlutterFlow/flutterflow-issues/assets/155745989/9bcbb66c-190f-4d1a-bec7-f6a403c6547b">

Any updates? I'm getting this when I deploy to web

**Comment by rzambroni** [comment]
Hey, sorry about that, we are currently investigating, the issue you are describing is not replicable in blank projects so it might be specific to something specific in your projects. You both have different errors as well.

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/2ea07307-27e1-44fb-9ffe-f6f5b459c00b/2beb7def-ce34-4487-844f-35b44e64fea7?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS8yZWEwNzMwNy0yN2UxLTQ0ZmItOWZmZS1mNmY1YjQ1OWMwMGIvMmJlYjdkZWYtY2UzNC00NDg3LTg0NGYtMzViNDRlNjRmZWE3IiwiaWF0IjoxNzE5NDA4ODE2LCJleHAiOjMzMjg5OTY4ODE2fQ.gxpZM-F4BhwImFfn3y2yLt9Muk6A6lvikw_m04EhkZ0 " alt="image.png" width="622" height="326" />

**Comment by ValetCEO** [comment]
So it looks like it's this pubspec dependancy that's failing: 

<img width="1680" alt="Screenshot 2024-06-26 at 14 43 43" src="https://github.com/FlutterFlow/flutterflow-issues/assets/155745989/f0010b47-7396-47d7-aebd-2e6afcda3527">

**Comment by rzambroni** [comment]
dependencies in custom code might need to be fixed due to the recent upgrade to Flutter 3.22.2 within FlutterFlow. And yes that could be the reason for the error.

**Comment by rzambroni** [comment]
In case it is helpful here is a quick guide on how to fix dependency issues: [https://www.youtube.com/watch?v=8Y1uyCC_dXE](https://www.youtube.com/watch?v=8Y1uyCC_dXE)

**Comment by ValetCEO** [comment]
<img width="1680" alt="Screenshot 2024-06-26 at 15 28 06" src="https://github.com/FlutterFlow/flutterflow-issues/assets/155745989/3a743e53-10b4-4114-bfb1-9d4bf3d2511e">

I've updated the dependencies and it now will deploy to web but I just get a blank screen. When I run it in test mode I just get this

**Comment by Esnoan** [comment]
> <img alt="Screenshot 2024-06-26 at 15 28 06" width="1680" src="https://private-user-images.githubusercontent.com/155745989/343204390-3a743e53-10b4-4114-bfb1-9d4bf3d2511e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0MTM0MDEsIm5iZiI6MTcxOTQxMzEwMSwicGF0aCI6Ii8xNTU3NDU5ODkvMzQzMjA0MzkwLTNhNzQzZTUzLTEwYjQtNDExNC1iZmIxLTlkNGJmM2QyNTExZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNlQxNDQ1MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYjgyNWI4MzQwYzk5YjU3NWU1ZDZjZTFkYzdjNjJiMTM3OGQ0YTExYmNkMDEyZDcxZTkyYmY0NjljYWU2MDA2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.WO18txoR2I-CqHtCmjqBEJIXcqKEeL7dGDUd6yhLXGk">
> I've updated the dependencies and it now will deploy to web but I just get a blank screen. When I run it in test mode I just get this

I have the same issue with error "Rejecting promise with error: TypeError: Cannot set properties of undefined (setting 'nativeCommunication')"

**Comment by ValetCEO** [comment]
Is anyone available to help with this ???? This has completely broken our platform. I've followed all the guides and it now deploys but just has a blank screen. We're live with paying users so this is a huge issue and I'm not getting any reply from the Flutterflow support team!

**Comment by ValetCEO** [comment]
<img width="1680" alt="Screenshot 2024-06-26 at 16 24 59" src="https://github.com/FlutterFlow/flutterflow-issues/assets/155745989/765ef501-0add-46dd-adee-9dc363667960">

Just getting a blank test screen now so no idea whats breaking or why and getting no response from the FF team.....fantastic

**Comment by rzambroni** [comment]
There are likely still custom code issues in your project. Have you been able to reach out to support? I know it might take a bit longer to get in touch but that's the best and most efficient way to get some help to fix these issues specific to each project.

**Comment by ValetCEO** [comment]
I've tried to contact support but no reply. If there were issues in the code it usually highlights them. I've updated the dependencies and followed their instructions but still nothing.

**Comment by Esnoan** [comment]
the error is caused by the package html_editor_enhanced, after removing it the app runs normally ☹️

**Comment by ValetCEO** [comment]
Correct, but after updating the dependency the app deploys but just shows a blank screen and if I revert to an earlier dependency it breaks again

**Comment by rsstevearend** [comment]
I'm having the same issue.  Support is not responding

**Comment by mriduljainmj** [comment]
I am also facing issues related to dependencies.
Let me use the correct version of WEBRTC which is compatible with it and then use it.
Will update if I get anything.

<img width="625" alt="Screenshot 2024-06-26 at 10 20 16 PM" src="https://github.com/FlutterFlow/flutterflow-issues/assets/57504232/8f58e0a8-ce4c-4ee2-8056-54a3afafb310">

**Comment by wanderodrigues** [comment]
We solved it by updating two libraries: infinite_scroll_pagination: ^4.0.0 and fl_chart: ^0.68.0.

**Comment by ValetCEO** [comment]
9hrs still no response from FF

**Comment by bloemy7** [comment]
We are running into the same issue and blocked by this. Please help.

**Comment by rsstevearend** [comment]
Same continued to be blocked by this issue.

**Comment by ValetCEO** [comment]
Is anyone from Flutterflow reading these messages ?

**Comment by ValetCEO** [comment]
We think we've identified the issue - It's the webview widget which is failing

**Comment by rsstevearend** [comment]
I'm seeing the exact same issue as you ValetCEO.  When I finally chatted with support they told me to watch this gitHub page.  Not very comforting since we are now on hold until this is resolved but no ETA.  Frustrating.

**Comment by Esnoan** [comment]
The bug is caused by flutter_inappwebview

The "solution" is add this script in the custom headers:
```
<script
      type="application/javascript"
      src="/assets/packages/flutter_inappwebview_web/assets/web/web_support.js"
      defer
    ></script>
```

you can find this script in the [docs](https://inappwebview.dev/docs/intro/#setup-web).

but this only works in local mode, in Test mode the app does not find the script. So I copied the content of this script and pasted it into the <script></script> tags. 

```
<script type="application/javascript" defer>
window.flutter_inappwebview = {
    webViews: {},
    createFlutterInAppWebView: function(viewId, iframeId) {
        var webView = {
            viewId: viewId,
            iframeId: iframeId,
            iframe: null,
            iframeContainer: null,
            windowAutoincrementId: 0,
            windows: {},
            isFullscreen: false,
            documentTitle: null,
            functionMap: {},
            settings: {},
            disableContextMenuHandler: function(event) {
                event.preventDefault();
                event.stopPropagation();
                return false;
            },
            prepare: function(settings) {
                webView.settings = settings;
                var iframe = document.getElementById(iframeId);
                var iframeContainer = document.getElementById(iframeId + '-container');

                document.addEventListener('fullscreenchange', function(event) {
                    // document.fullscreenElement will point to the element that
                    // is in fullscreen mode if there is one. If there isn't one,
                    // the value of the property is null.
                    if (document.fullscreenElement && document.fullscreenElement.id == iframeId) {
                        webView.isFullscreen = true;
                        window.flutter_inappwebview.nativeCommunication('onEnterFullscreen', viewId);
                    } else if (!document.fullscreenElement && webView.isFullscreen) {
                        webView.isFullscreen = false;
                        window.flutter_inappwebview.nativeCommunication('onExitFullscreen', viewId);
                    } else {
                        webView.isFullscreen = false;
                    }
                });

                if (iframe != null) {
                    webView.iframe = iframe;
                    webView.iframeContainer = iframeContainer;
                    iframe.addEventListener('load', function (event) {
                        webView.windowAutoincrementId = 0;
                        webView.windows = {};

                        var url = iframe.src;
                        try {
                            url = iframe.contentWindow.location.href;
                        } catch (e) {
                            console.log(e);
                        }
                        window.flutter_inappwebview.nativeCommunication('onLoadStart', viewId, [url]);

                        try {
                            var oldLogs = {
                                'log': iframe.contentWindow.console.log,
                                'debug': iframe.contentWindow.console.debug,
                                'error': iframe.contentWindow.console.error,
                                'info': iframe.contentWindow.console.info,
                                'warn': iframe.contentWindow.console.warn
                            };
                            for (var k in oldLogs) {
                                (function(oldLog) {
                                    iframe.contentWindow.console[oldLog] = function() {
                                        var message = '';
                                        for (var i in arguments) {
                                            if (message == '') {
                                                message += arguments[i];
                                            } else {
                                                message += ' ' + arguments[i];
                                            }
                                        }
                                        oldLogs[oldLog].call(iframe.contentWindow.console, ...arguments);
                                        window.flutter_inappwebview.nativeCommunication('onConsoleMessage', viewId, [oldLog, message]);
                                    }
                                })(k);
                            }
                        } catch (e) {
                            console.log(e);
                        }

                        try {
                            var originalPushState = iframe.contentWindow.history.pushState;
                            iframe.contentWindow.history.pushState = function (state, unused, url) {
                                originalPushState.call(iframe.contentWindow.history, state, unused, url);
                                var iframeUrl = iframe.src;
                                try {
                                    iframeUrl = iframe.contentWindow.location.href;
                                } catch (e) {
                                    console.log(e);
                                }
                                window.flutter_inappwebview.nativeCommunication('onUpdateVisitedHistory', viewId, [iframeUrl]);
                            };

                            var originalReplaceState = iframe.contentWindow.history.replaceState;
                            iframe.contentWindow.history.replaceState = function (state, unused, url) {
                                originalReplaceState.call(iframe.contentWindow.history, state, unused, url);
                                var iframeUrl = iframe.src;
                                try {
                                    iframeUrl = iframe.contentWindow.location.href;
                                } catch (e) {
                                    console.log(e);
                                }
                                window.flutter_inappwebview.nativeCommunication('onUpdateVisitedHistory', viewId, [iframeUrl]);
                            };

                            var originalOpen = iframe.contentWindow.open;
                            iframe.contentWindow.open = function (url, target, windowFeatures) {
                                var newWindow = originalOpen.call(iframe.contentWindow, ...arguments);
                                var windowId = webView.windowAutoincrementId;
                                webView.windowAutoincrementId++;
                                webView.windows[windowId] = newWindow;
                                window.flutter_inappwebview.nativeCommunication('onCreateWindow', viewId, [windowId, url, target, windowFeatures]).then(function(){}, function(handledByClient) {
                                    if (handledByClient) {
                                        newWindow.close();
                                    }
                                });
                                return newWindow;
                            };

                            var originalPrint = iframe.contentWindow.print;
                            iframe.contentWindow.print = function() {
                                var iframeUrl = iframe.src;
                                try {
                                    iframeUrl = iframe.contentWindow.location.href;
                                } catch (e) {
                                    console.log(e);
                                }
                                window.flutter_inappwebview.nativeCommunication('onPrintRequest', viewId, [iframeUrl]);
                                originalPrint.call(iframe.contentWindow);
                            };

                            webView.functionMap = {
                                "window.open": iframe.contentWindow.open,
                                "window.print": iframe.contentWindow.print,
                                "window.history.pushState": iframe.contentWindow.history.pushState,
                                "window.history.replaceState": iframe.contentWindow.history.replaceState,
                            }

                            var initialTitle = iframe.contentDocument.title;
                            var titleEl = iframe.contentDocument.querySelector('title');
                            webView.documentTitle = initialTitle;
                            window.flutter_inappwebview.nativeCommunication('onTitleChanged', viewId, [initialTitle]);
                            if (titleEl != null) {
                                new MutationObserver(function(mutations) {
                                    var title = mutations[0].target.innerText;
                                    if (title != webView.documentTitle) {
                                        webView.documentTitle = title;
                                        window.flutter_inappwebview.nativeCommunication('onTitleChanged', viewId, [title]);
                                    }
                                }).observe(
                                    titleEl,
                                    { subtree: true, characterData: true, childList: true }
                                );
                            }

                            var oldPixelRatio = iframe.contentWindow.devicePixelRatio;
                            iframe.contentWindow.addEventListener('resize', function (e) {
                                var newPixelRatio = iframe.contentWindow.devicePixelRatio;
                                if(newPixelRatio !== oldPixelRatio){
                                    window.flutter_inappwebview.nativeCommunication('onZoomScaleChanged', viewId, [oldPixelRatio, newPixelRatio]);
                                    oldPixelRatio = newPixelRatio;
                                }
                            });

                            iframe.contentWindow.addEventListener('popstate', function (event) {
                                var iframeUrl = iframe.src;
                                try {
                                    iframeUrl = iframe.contentWindow.location.href;
                                } catch (e) {
                                    console.log(e);
                                }
                                window.flutter_inappwebview.nativeCommunication('onUpdateVisitedHistory', viewId, [iframeUrl]);
                            });

                            iframe.contentWindow.addEventListener('scroll', function (event) {
                                var x = 0;
                                var y = 0;
                                try {
                                    x = iframe.contentWindow.scrollX;
                                    y = iframe.contentWindow.scrollY;
                                } catch (e) {
                                    console.log(e);
                                }
                                window.flutter_inappwebview.nativeCommunication('onScrollChanged', viewId, [x, y]);
                            });

                            iframe.contentWindow.addEventListener('focus', function (event) {
                                window.flutter_inappwebview.nativeCommunication('onWindowFocus', viewId);
                            });

                            iframe.contentWindow.addEventListener('blur', function (event) {
                                window.flutter_inappwebview.nativeCommunication('onWindowBlur', viewId);
                            });
                        } catch (e) {
                            console.log(e);
                        }

                        try {

                            if (!webView.settings.javaScriptCanOpenWindowsAutomatically) {
                                iframe.contentWindow.open = function() {
                                    throw new Error('JavaScript cannot open windows automatically');
                                };
                            }

                            if (!webView.settings.verticalScrollBarEnabled && !webView.settings.horizontalScrollBarEnabled) {
                                var style = iframe.contentDocument.createElement('style');
                                style.id = "settings.verticalScrollBarEnabled-settings.horizontalScrollBarEnabled";
                                style.innerHTML = "body::-webkit-scrollbar { width: 0px; height: 0px; }";
                                iframe.contentDocument.head.append(style);
                            }

                            if (webView.settings.disableVerticalScroll) {
                                var style = iframe.contentDocument.createElement('style');
                                style.id = "settings.disableVerticalScroll";
                                style.innerHTML = "body { overflow-y: hidden; }";
                                iframe.contentDocument.head.append(style);
                            }

                            if (webView.settings.disableHorizontalScroll) {
                                var style = iframe.contentDocument.createElement('style');
                                style.id = "settings.disableHorizontalScroll";
                                style.innerHTML = "body { overflow-x: hidden; }";
                                iframe.contentDocument.head.append(style);
                            }

                            if (webView.settings.disableContextMenu) {
                                iframe.contentWindow.addEventListener('contextmenu', webView.disableContextMenuHandler);
                            }
                        } catch (e) {
                            console.log(e);
                        }

                        window.flutter_inappwebview.nativeCommunication('onLoadStop', viewId, [url]);
                    });
                }
            },
            setSettings: function(newSettings) {
                var iframe = webView.iframe;
                try {
                    if (webView.settings.javaScriptCanOpenWindowsAutomatically != newSettings.javaScriptCanOpenWindowsAutomatically) {
                        if (!newSettings.javaScriptCanOpenWindowsAutomatically) {
                            iframe.contentWindow.open = function() {
                                throw new Error('JavaScript cannot open windows automatically');
                            };
                        } else {
                            iframe.contentWindow.open = webView.functionMap["window.open"];
                        }
                    }

                    if (webView.settings.verticalScrollBarEnabled != newSettings.verticalScrollBarEnabled &&
                        webView.settings.horizontalScrollBarEnabled != newSettings.horizontalScrollBarEnabled) {
                        if (!newSettings.verticalScrollBarEnabled && !newSettings.horizontalScrollBarEnabled) {
                            var style = iframe.contentDocument.createElement('style');
                            style.id = "settings.verticalScrollBarEnabled-settings.horizontalScrollBarEnabled";
                            style.innerHTML = "body::-webkit-scrollbar { width: 0px; height: 0px; }";
                            iframe.contentDocument.head.append(style);
                        } else {
                            var styleElement = iframe.contentDocument.getElementById("settings.verticalScrollBarEnabled-settings.horizontalScrollBarEnabled");
                            if (styleElement) { styleElement.remove() }
                        }
                    }

                    if (webView.settings.disableVerticalScroll != newSettings.disableVerticalScroll) {
                        if (newSettings.disableVerticalScroll) {
                            var style = iframe.contentDocument.createElement('style');
                            style.id = "settings.disableVerticalScroll";
                            style.innerHTML = "body { overflow-y: hidden; }";
                            iframe.contentDocument.head.append(style);
                        } else {
                             var styleElement = iframe.contentDocument.getElementById("settings.disableVerticalScroll");
                            if (styleElement) { styleElement.remove() }
                        }
                    }

                    if (webView.settings.disableHorizontalScroll != newSettings.disableHorizontalScroll) {
                        if (newSettings.disableHorizontalScroll) {
                            var style = iframe.contentDocument.createElement('style');
                            style.id = "settings.disableHorizontalScroll";
                            style.innerHTML = "body { overflow-x: hidden; }";
                            iframe.contentDocument.head.append(style);
                        } else {
                             var styleElement = iframe.contentDocument.getElementById("settings.disableHorizontalScroll");
                            if (styleElement) { styleElement.remove() }
                        }
                    }

                    if (webView.settings.disableContextMenu != newSettings.disableContextMenu) {
                        if (newSettings.disableContextMenu) {
                            iframe.contentWindow.addEventListener('contextmenu', webView.disableContextMenuHandler);
                        } else {
                            iframe.contentWindow.removeEventListener('contextmenu', webView.disableContextMenuHandler);
                        }
                    }
                } catch (e) {
                    console.log(e);
                }

                webView.settings = newSettings;
            },
            reload: function() {
                var iframe = webView.iframe;
                if (iframe != null && iframe.contentWindow != null) {
                    try {
                        iframe.contentWindow.location.reload();
                    } catch (e) {
                        console.log(e);
                        iframe.contentWindow.location.href = iframe.src;
                    }
                }
            },
            goBack: function() {
                var iframe = webView.iframe;
                if (iframe != null) {
                    try {
                        iframe.contentWindow.history.back();
                    } catch (e) {
                        console.log(e);
                    }
                }
            },
            goForward: function() {
                var iframe = webView.iframe;
                if (iframe != null) {
                    try {
                        iframe.contentWindow.history.forward();
                    } catch (e) {
                        console.log(e);
                    }
                }
            },
            goBackOrForward: function(steps) {
                var iframe = webView.iframe;
                if (iframe != null) {
                    try {
                        iframe.contentWindow.history.go(steps);
                    } catch (e) {
                        console.log(e);
                    }
                }
            },
            evaluateJavascript: function(source) {
                var iframe = webView.iframe;
                var result = null;
                if (iframe != null) {
                    try {
                        result = JSON.stringify(iframe.contentWindow.eval(source));
                    } catch (e) {}
                }
                return result;
            },
            stopLoading: function(steps) {
                var iframe = webView.iframe;
                if (iframe != null) {
                    try {
                        iframe.contentWindow.stop();
                    } catch (e) {
                        console.log(e);
                    }
                }
            },
            getUrl: function() {
                var iframe = webView.iframe;
                var url = iframe.src;
                try {
                    url = iframe.contentWindow.location.href;
                } catch (e) {
                    console.log(e);
                }
                return url;
            },
            getTitle: function() {
                var iframe = webView.iframe;
                var title = null;
                try {
                    title = iframe.contentDocument.title;
                } catch (e) {
                    console.log(e);
                }
                return title;
            },
            injectJavascriptFileFromUrl: function(urlFile, scriptHtmlTagAttributes) {
                var iframe = webView.iframe;
                try {
                    var d = iframe.contentDocument;
                    var script = d.createElement('script');
                    for (var key of Object.keys(scriptHtmlTagAttributes)) {
                        if (scriptHtmlTagAttributes[key] != null) {
                            script[key] = scriptHtmlTagAttributes[key];
                        }
                    }
                    if (script.id != null) {
                        script.onload = function() {
                            window.flutter_inappwebview.nativeCommunication('onInjectedScriptLoaded', webView.viewId, [script.id]);
                        }
                        script.onerror = function() {
                            window.flutter_inappwebview.nativeCommunication('onInjectedScriptError', webView.viewId, [script.id]);
                        }
                    }
                    script.src = urlFile;
                    if (d.body != null) {
                        d.body.appendChild(script);
                    }
                } catch (e) {
                    console.log(e);
                }
            },
            injectCSSCode: function(source) {
                var iframe = webView.iframe;
                try {
                    var d = iframe.contentDocument;
                    var style = d.createElement('style');
                    style.innerHTML = source;
                    if (d.head != null) {
                        d.head.appendChild(style);
                    }
                } catch (e) {
                    console.log(e);
                }
            },
            injectCSSFileFromUrl: function(urlFile, cssLinkHtmlTagAttributes) {
                var iframe = webView.iframe;
                try {
                    var d = iframe.contentDocument;
                    var link = d.createElement('link');
                    for (var key of Object.keys(cssLinkHtmlTagAttributes)) {
                        if (cssLinkHtmlTagAttributes[key] != null) {
                            link[key] = cssLinkHtmlTagAttributes[key];
                        }
                    }
                    link.type = 'text/css';
                    var alternateStylesheet = "";
                    if (cssLinkHtmlTagAttributes.alternateStylesheet) {
                        alternateStylesheet = "alternate ";
                    }
                    link.rel = alternateStylesheet + "stylesheet";
                    link.href = urlFile;
                    if (d.head != null) {
                        d.head.appendChild(link);
                    }
                } catch (e) {
                    console.log(e);
                }
            },
            scrollTo: function(x, y, animated) {
                var iframe = webView.iframe;
                try {
                    if (animated) {
                        iframe.contentWindow.scrollTo({top: y, left: x, behavior: 'smooth'});
                    } else {
                        iframe.contentWindow.scrollTo(x, y);
                    }
                } catch (e) {
                    console.log(e);
                }
            },
            scrollBy: function(x, y, animated) {
                var iframe = webView.iframe;
                try {
                    if (animated) {
                        iframe.contentWindow.scrollBy({top: y, left: x, behavior: 'smooth'});
                    } else {
                        iframe.contentWindow.scrollBy(x, y);
                    }
                } catch (e) {
                    console.log(e);
                }
            },
            printCurrentPage: function() {
                var iframe = webView.iframe;
                try {
                    iframe.contentWindow.print();
                } catch (e) {
                    console.log(e);
                }
            },
            getContentHeight: function() {
                var iframe = webView.iframe;
                try {
                    return iframe.contentDocument.documentElement.scrollHeight;
                } catch (e) {
                    console.log(e);
                }
                return null;
            },
            getContentWidth: function() {
                var iframe = webView.iframe;
                try {
                    return iframe.contentDocument.documentElement.scrollWidth;
                } catch (e) {
                    console.log(e);
                }
                return null;
            },
            getSelectedText: function() {
                var iframe = webView.iframe;
                try {
                    var txt;
                    var w = iframe.contentWindow;
                    if (w.getSelection) {
                        txt = w.getSelection().toString();
                    } else if (w.document.getSelection) {
                        txt = w.document.getSelection().toString();
                    } else if (w.document.selection) {
                        txt = w.document.selection.createRange().text;
                    }
                    return txt;
                } catch (e) {
                    console.log(e);
                }
                return null;
            },
            getScrollX: function() {
                var iframe = webView.iframe;
                try {
                    return iframe.contentWindow.scrollX;
                } catch (e) {
                    console.log(e);
                }
                return null;
            },
            getScrollY: function() {
                var iframe = webView.iframe;
                try {
                    return iframe.contentWindow.scrollY;
                } catch (e) {
                    console.log(e);
                }
                return null;
            },
            isSecureContext: function() {
                var iframe = webView.iframe;
                try {
                    return iframe.contentWindow.isSecureContext;
                } catch (e) {
                    console.log(e);
                }
                return false;
            },
            canScrollVertically: function() {
                var iframe = webView.iframe;
                try {
                    return iframe.contentDocument.body.scrollHeight > iframe.contentWindow.innerHeight;
                } catch (e) {
                    console.log(e);
                }
                return false;
            },
            canScrollHorizontally: function() {
                var iframe = webView.iframe;
                try {
                    return iframe.contentDocument.body.scrollWidth > iframe.contentWindow.innerWidth;
                } catch (e) {
                    console.log(e);
                }
                return false;
            },
            getSize: function() {
                var iframeContainer = webView.iframeContainer;
                var width = 0.0;
                var height = 0.0;
                if (iframeContainer.style.width != null && iframeContainer.style.width != '' && iframeContainer.style.width.indexOf('px') > 0) {
                    width = parseFloat(iframeContainer.style.width);
                }
                if (width == null || width == 0.0) {
                    width = iframeContainer.getBoundingClientRect().width;
                }
                if (iframeContainer.style.height != null && iframeContainer.style.height != '' && iframeContainer.style.height.indexOf('px') > 0) {
                    height = parseFloat(iframeContainer.style.height);
                }
                if (height == null || height == 0.0) {
                    height = iframeContainer.getBoundingClientRect().height;
                }

                return {
                    width: width,
                    height: height
                };
            }
        };

        return webView;
    },
    getCookieExpirationDate: function(timestamp) {
        return (new Date(timestamp)).toUTCString();
    }
};
</script>
```

if anyone has a better solution, please share it.

**Comment by sdschroeder** [comment]
Same blank page issue here. Our web app is mission critical and I have no idea where to look. All dependencies show no issues in the admin.

**Comment by sdschroeder** [comment]
Even though I wasn't seeing any issues with dependencies, I went ahead and updated everything anyways without success.

But, since I was using the HTMLEditorEnhanced, I went ahead and deleted all references to that and it solved my issue. Disappointed I can't have an HTML editor in my apps anymore though.

It would be great to know how to make it possible to use again...

**Comment by ValetCEO** [comment]
My app is unusable without the rich text editor. This is INSANE !!!! 3 messages a day I get from the FF support team, none of them helpful and they seem like they just don't give a shit that they've broken so many peoples apps!

**Comment by mriduljainmj** [comment]
Soo True !!! I am trying it from the past 2 days but a error from the PUB cache from the previous version is creating some issue and thats why i am not able to deploy it on web. Now I am also not able to update the version of it also.

**Comment by rzambroni** [comment]
Hey, there is another fix to the custom code dependency issue that was discussed on the following ticket: [https://github.com/FlutterFlow/flutterflow-issues/issues/3250#issuecomment-2197274879](https://github.com/FlutterFlow/flutterflow-issues/issues/3250#issuecomment-2197274879)

And proved to be helpful to some people. 

Hope that helps.

**Comment by mriduljainmj** [comment]
Is there any way to clear the cached dependencies of flutter while publishing them.

**Comment by edualvesilva2** [comment]
I updated html_editor_enhanced: ^2.6.0 to the latest version, compatible with flutter 3.22, and despite not having any errors in the code, my application does not run on the web version, the screen appears blank. 

![Sem título-1 fw](https://github.com/FlutterFlow/flutterflow-issues/assets/6314150/1d7dd1ba-d9df-4255-8500-9107e6fa79de)
Here are the errors in the print:

**Comment by jakubmichalski** [comment]
Following [advice](https://github.com/FlutterFlow/flutterflow-issues/issues/3229#issuecomment-2195241746) from @Esnoan has worked for me with the latest HTML editor version 2.6.0. I mention it again here as I see it might have been missed through the conversations.
